### ISSUE TITLE
Database updates need hours on OSMC / Raspberry Pi. Solution: directly use MediathekViewWeb API

### ISSUE THREAD

**Issue by Wikinaut** [mediathekview/plugin.video.mediathekview#50]
I run the mediathekview plugin since it has been published, it takes one hour or more to update the database.

Why are you not dropping fully the local databases business and query directly the wonderful MediathekViewWeb like https://mediathekviewweb.de/#query=test ?

**Comment by tuxpoldo** [comment]
1. This Kodi Plugin intends to became an offline capable solution (when batch download is fully implemented)
1. There is no real documentation available about the MediathekViewWeb API - or at least I do not know about it... Also I have no guarantee that things will change...
1. If your RasPI takes one and a half hours, something must be wrong. We have tested it on several RasPI and we have full update times around 10-15 minutes. No idea if it is caused by your hardware (maybe SD card?) or OSMC.

Maybe in future there will be additional backend database drivers...

**Comment by tuxpoldo** [comment]
I will reopen this because the idea is interesting and at least I want to discuss it more in details with some people from the project....

**Comment by Wikinaut** [comment]
@tuxpoldo Yes, thanks. My (basic) idea was to simply have the same "command line" as on his web page. And then let OSMC (simply) print the line/s of results.

I am now pretty sure, that you understand my idea. Pls. feel free to change(adapt) the issue title, if needed.

**Comment by krassle** [comment]
First DB creation on a FireTV Stick (1st Gen.) took around 20 minutes. Diff. updates are much faster ~ 1-3min. The only problem I have ATM is browsing speed, especially of some folders( like 'Live Streams' ) it takes up to 30sec. just to get in. Is this DB related(too much queries at once)?  Playback (starts and stops) is a bit slow too.

**Comment by derandiunddasbo** [comment]
I'd like to support the web API feature request as well. On my RPi3 even the diff updates take several minutes and browsing speed is very slow for large result lists. 

I really like the idea of the Kodi port of Mediathekview, as it could replace a lot of other Mediathek addons.

But most of the times I run Kodi once a day to watch a TV show. But unfortunately the plugin is virtually unusable until it finished it's daily database update and even if it's finished, it often takes 30 to 60 seconds to open a folder or category.

To be honest, I only used the plugin two times since I installed it weeks ago, because most of the time I lose my patience waiting for the plugin to finish it's updates and simply cast the show I want to see from an Android device to a chromecast connected to the same TV.
